mirrord/intproxy/src/proxies/simple.rs

```diff
@@ -27,20 +32,40 @@ pub enum SimpleProxyMessage {
     LayerClosed(LayerClosed),
     GetEnvReq(MessageId, LayerId, GetEnvVarsRequest),
     GetEnvRes(RemoteResult<HashMap<String, String>>),
+    ProtocolVersion(Version),
 }
 
 #[derive(Clone, Copy, PartialEq, Eq, Hash)]
-enum RemoteFd {
+pub(crate) enum RemoteFd {
     File(u64),
     Dir(u64),
 }
 
+#[derive(Clone)]
+pub(crate) enum FileResource {
+    File,
+    Dir {
+        dirs_iter: Option<IntoIter<DirEntryInternal>>,
+    },
+}
+
+impl FileResource {
+    fn next_dir(&mut self) -> Option<DirEntryInternal> {
```

Comment:
This should return an error when called on a regular file

Reply:
Changed it to return proper errors when we hit an error in the cached readdir.